### PR TITLE
Replacements

### PR DESCRIPTION
**Pull request by jeromemacias**
Added possibility to make replacements after decoding the configuration file.

### CODE REVIEW

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 1]
```diff
@@ -38,7 +46,23 @@ class ConfigServiceProvider implements ServiceProviderInterface
...
         }
+
+        if (is_array($value))
+        {
+            foreach ($value as $k => $v) {
+                $value[$k] = $this->doReplacements($v);
+            }
+        }
+
+        return str_replace(array_keys($this->replacements), array_values($this->replacements), $value);
```
Comment (igorw) [inline src/Igorw/Silex/ConfigServiceProvider.php]: You can do:

```
return strtr($value, $this->replacements);
```

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 2]
```diff
@@ -38,7 +46,23 @@ class ConfigServiceProvider implements ServiceProviderInterface
...
+        }
+    }
+
+    private function doReplacements($value)
+    {
+        if (!$this->replacements) {
+            return $value;
         }
+
+        if (is_array($value))
```
Comment (igorw) [inline src/Igorw/Silex/ConfigServiceProvider.php]: Opening brace should be on the same line as `if (`.

File: README.md [hunk 3]
```diff
@@ -39,3 +39,18 @@ that would be:
...
+    ));
+
+Now you can use the pattern in your configuration file.
+
+**/config/services.json**
+
+    {
+        "db.dbal.class_path": "%dbalpath%",
+        "db.common.class_path": "%dbalpath%/vendor/doctrine-common/lib",
+    }
```
Comment (igorw) [inline README.md:55]: I don't really like this example. I'm working on a solution that the class paths no longer have to be specified. Also, the JSON is invalid, there should not be a trailing comma.

Do you have any other use cases that could be used as an example?

**Comment by igorw** [comment]
Awesome, thanks!

**Comment by docteurklein** [comment]
cool one!

**Pull request merged**